account_block_account/models/blocked_accounts_mixin.py

```diff
@@ -0,0 +1,38 @@
+# -*- coding: utf-8 -*-
+# © 2018 Therp BV <https://therp.nl>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+from openerp import _, api, models
+from openerp.exceptions import ValidationError
+
+
+class BlockedAccountsMixin(models.AbstractModel):
+    """Inherit this for supporting blocks on accounts"""
+    _name = 'blocked.accounts.mixin'
+    _description = 'A model that understands about blocked accounts'
+
+    # set this to the field(s) pointing to models which can be blocked
+    _blocked_accounts_fields = []
+
+    @api.model
+    def create(self, values):
+        result = super(BlockedAccountsMixin, self).create(values)
+        result._assert_non_blocked(values)
+        return result
+
+    @api.multi
+    def write(self, values):
+        result = super(BlockedAccountsMixin, self).write(values)
+        self._assert_non_blocked(values)
+        return result
+
+    @api.multi
+    def _assert_non_blocked(self, values):
+        fields_list = set(values) & set(self._fields) & set(
```

Comment:
What is the rationale behind `self._fields` here? Wont that include all fields?

Reply:
it's a conjunction. You're feverish after all.

This restriction makes it simple for programmers to use the mixin, because whatever they fuck up in `self._blocked_accounts_fields` will be filtered out by this sanity check

Reply:
also if `values` had some invalid key, we'd raise below. maybe we should

Reply:
Ah, never knew. This is good then. Raising is a good idea to help people debug where they went wrong.